src/react/AppProvider.jsx

```diff
@@ -63,7 +68,23 @@ export default function AppProvider({ store, children }) {
     setLocale(getLocale());
   });
 
-  const appContextValue = useMemo(() => ({ authenticatedUser, config, locale }), [authenticatedUser, config, locale]);
+  const [paragonThemeState, paragonThemeDispatch] = useParagonTheme(config);
+
+  const appContextValue = useMemo(() => ({
+    authenticatedUser,
+    config,
+    locale,
+    paragonTheme: {
+      state: paragonThemeState,
+      setThemeVariant: (themeVariant) => {
+        paragonThemeDispatch(paragonThemeActions.setParagonThemeVariant(themeVariant));
+      },
+    },
+  }), [authenticatedUser, config, locale, paragonThemeState, paragonThemeDispatch]);
```

Comment:
Same sanity check question for `paragonThemeState`, I guess, though that'd be shocking if it weren't.

Reply:
Both `paragonThemeState` and `dispatch` from `useReducer` are safe to use in dependency arrays (memoized already like a state setter function).

From https://react.dev/reference/react/useReducer#usereducer:

> **If the new value you provide is identical to the current state, as determined by an [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will skip re-rendering the component and its children.** This is an optimization. React may still need to call your component before ignoring the result, but it shouldn’t affect your code.

From https://shopify.engineering/master-reacts-stable-values:

> useState and **useReducer return a state update function that is constant** – the hook will always return the same function. The value is guaranteed to be stable, because we know that the same object will be returned until we call the state update function.